### ISSUE TITLE
AttributeError: 'NavRoute' object has no attribute 'target'

### ISSUE THREAD

**Issue by hawesie** [strands-project/strands_navigation#206]
I'm trying to integrate the latest policy execution stuff into the scheduler and executor. 

Looks like the NavRoute message is missing an attribute which is expected in the code in `execute_policy_server.py`. 

```
Do move_base to WayPoint7
[ERROR] [WallTime: 1427404269.180533] Exception in your execute callback: global name 'route' is not defined
Traceback (most recent call last):
  File "/opt/ros/indigo/lib/python2.7/site-packages/actionlib/simple_action_server.py", line 299, in executeLoop
    self.execute_callback(goal)
  File "/Users/nah/code/indigo_ws/src/strands_navigation/topological_navigation/scripts/execute_policy_server.py", line 180, in executeCallback
    result = self.followRoute(goal.route)
  File "/Users/nah/code/indigo_ws/src/strands_navigation/topological_navigation/scripts/execute_policy_server.py", line 223, in followRoute
    result=self.execute_policy(route)
  File "/Users/nah/code/indigo_ws/src/strands_navigation/topological_navigation/scripts/execute_policy_server.py", line 335, in execute_policy
    success=self.navigate_to(self.current_action,self.current_node)
  File "/Users/nah/code/indigo_ws/src/strands_navigation/topological_navigation/scripts/execute_policy_server.py", line 386, in navigate_to
    edg= self.get_edge_id(route[rindex].name, route[rindex+1].name, action)
NameError: global name 'route' is not defined

WayPoint7 -(move_base)-> WayPoint8
[ERROR] [WallTime: 1427404270.923653] Exception in your execute callback: global name 'route' is not defined
Traceback (most recent call last):
  File "/opt/ros/indigo/lib/python2.7/site-packages/actionlib/simple_action_server.py", line 299, in executeLoop
    self.execute_callback(goal)
  File "/Users/nah/code/indigo_ws/src/strands_navigation/topological_navigation/scripts/execute_policy_server.py", line 180, in executeCallback
    result = self.followRoute(goal.route)
  File "/Users/nah/code/indigo_ws/src/strands_navigation/topological_navigation/scripts/execute_policy_server.py", line 223, in followRoute
    result=self.execute_policy(route)
  File "/Users/nah/code/indigo_ws/src/strands_navigation/topological_navigation/scripts/execute_policy_server.py", line 256, in execute_policy
    success=self.navigate_to(self.current_action, target)
  File "/Users/nah/code/indigo_ws/src/strands_navigation/topological_navigation/scripts/execute_policy_server.py", line 386, in navigate_to
    edg= self.get_edge_id(route[rindex].name, route[rindex+1].name, action)
NameError: global name 'route' is not defined

[ERROR] [WallTime: 1427404270.957804] bad callback: <bound method TopologicalMapVis.policies_callback of <__main__.TopologicalMapVis object at 0x10726ec90>>
Traceback (most recent call last):
  File "/opt/ros/indigo/lib/python2.7/site-packages/rospy/topics.py", line 702, in _invoke_callback
    cb(msg)
  File "/Users/nah/code/indigo_ws/src/strands_navigation/topological_navigation/scripts/visualise_map.py", line 122, in policies_callback
    self.policies.received_route(msg)
  File "/Users/nah/code/indigo_ws/src/strands_navigation/topological_navigation/src/topological_navigation/policies.py", line 96, in received_route
    self.update_map(self.topo_map)
  File "/Users/nah/code/indigo_ws/src/strands_navigation/topological_navigation/src/topological_navigation/policies.py", line 42, in update_map
    indt = self.topo_map._get_node_index(self.route_nodes.target[counter])
AttributeError: 'NavRoute' object has no attribute 'target'
```

**Comment by hawesie** [comment]
Either @Jailander or @bfalacerda should be assigned I guess.

**Comment by bfalacerda** [comment]
are you sure you have everything up-to-date and you don't have any old local workspace being sourced? Who is calling the topological_navigation action server? It shouldn't be called when you're executing policies. All those error messages look very weird...

**Comment by bfalacerda** [comment]
sorry, misread the errors. ignore last comment

**Comment by bfalacerda** [comment]
I guess the `NameError: global name 'route' is not defined` error is some issue with the new logging you added today @Jailander ?

**Comment by hawesie** [comment]
It looks like my code is up to date. The last commit I have for strands_navigation is f55e53dbc88b01ce16b2fa8960fb2cc3593bae5e

**Comment by hawesie** [comment]
Yep, it was in commit 72f03d3f36fe413316ed78655d3fb55482f9ec8d that the buggy line appears to have been added.

THIS IS WHY WE NEED UNIT TESTS @marc-hanheide

**Comment by Jailander** [comment]
Hi, yes that was a last minute addition I didn't test policy execution just top_nav solved by #207

**Comment by Jailander** [comment]
closing this
